application/src/main/java/org/thingsboard/server/service/queue/processing/AbstractTbRuleEngineSubmitStrategy.java

```diff
@@ -52,14 +52,12 @@ public void update(ConcurrentMap<UUID, TbProtoQueueMsg<TransportProtos.ToRuleEng
         List<IdMsgPair<TransportProtos.ToRuleEngineMsg>> newOrderedMsgList = new ArrayList<>(reprocessMap.size());
         for (IdMsgPair<TransportProtos.ToRuleEngineMsg> pair : orderedMsgList) {
             if (reprocessMap.containsKey(pair.uuid)) {
-                var oldValue = pair.getMsg().getValue();
-                if (StringUtils.isNotEmpty(oldValue.getFailureMessage())) {
-                    var newValue = TransportProtos.ToRuleEngineMsg.newBuilder()
-                            .setTenantIdMSB(oldValue.getTenantIdMSB())
-                            .setTenantIdLSB(oldValue.getTenantIdLSB())
-                            .setTbMsg(oldValue.getTbMsg())
+                if (StringUtils.isNotEmpty(pair.getMsg().getValue().getFailureMessage())) {
+                    var toRuleEngineMsg = TransportProtos.ToRuleEngineMsg.newBuilder(pair.getMsg().getValue())
+                            .clearFailureMessage()
+                            .clearRelationTypes()
```

Comment:
TY. That is much more descriptive than before! failures and relations will be cleared.